src/intrprtr.c

```diff
@@ -277,28 +233,23 @@ static void FinishAndCallFakeFuncExpr(Obj stackNams)
 **  return-void-statement was interpreted. If 'IntrEnd' returns 'STATUS_QUIT',
 **  then a quit-statement was interpreted.
 */
-void IntrBegin ( Obj frame )
+void IntrBegin(IntrState * intr,  Obj frame)
 {
-    /* remember old interpreter state                                      */
-    if (!STATE(IntrState))
-        STATE(IntrState) = NEW_PLIST(T_PLIST, 16);
-    PushPlist(STATE(IntrState), STATE(StackObj));
-
     /* allocate a new values stack                                         */
-    STATE(StackObj) = NEW_PLIST( T_PLIST, 64 );
+    intr->StackObj = NEW_PLIST( T_PLIST, 64 );
 
     /* must be in immediate (non-ignoring, non-coding) mode                */
-    assert( STATE(IntrIgnoring) == 0 );
-    assert( STATE(IntrCoding)   == 0 );
+    GAP_ASSERT(intr->ignoring == 0);
```

Comment:
Oh yeah, I changed a few `assert` to `GAP_ASSERT` since I was touching those lines anyway

Reply:
It's out own macro, under our own full control; it does what we want, not the compiler or OS vendor. For now it is an alias for `assert`, but only if kernel debug mode was activated by configure